src/app/profile/core.nim

```diff
@@ -55,6 +56,30 @@ proc init*(self: ProfileController, account: Account) =
   self.status.chat.updateContacts(contacts)
   self.view.setContactList(contacts)
 
+  self.status.events.on("channelLoaded") do(e: Args):
+    var channel = ChannelArgs(e)
+    if channel.chat.muted:
+      if channel.chat.chatType.isOneToOne:
+        discard self.view.mutedContacts.addChatItemToList(channel.chat)
+        return
+      discard self.view.mutedChats.addChatItemToList(channel.chat)
+
+  self.status.events.on("channelJoined") do(e: Args):
+    var channel = ChannelArgs(e)
+    if channel.chat.muted:
+      if channel.chat.chatType.isOneToOne:
+        discard self.view.mutedContacts.addChatItemToList(channel.chat)
+        return
+      discard self.view.mutedChats.addChatItemToList(channel.chat)
```

Comment:
Those are the same lines as above if I see correctly. You could extract them in a function to reduce duplication :)